src/rgw/rgw-orphan-list

```diff
@@ -57,19 +58,49 @@ echo "Pool is \"$pool\"."
 echo "Note: output files produced will be tagged with the current timestamp -- ${timestamp}."
 
 echo "running 'rados ls' at $(date)"
-rados ls --pool="$pool" --format=json-pretty --all >"$rados_out" 2>"$rados_err"
+# since --format is not specified, plain should be used
+rados ls --pool="$pool" --all >"$rados_out" 2>"$rados_err"
 if [ "$?" -ne 0 ] ;then
     error_out "rados ls" "$rados_err"
 fi
 
-# check for namespaces and error out if any found
-grep '^[[:blank:]]*"namespace":' "$rados_out" | grep --silent '[^:]*: "[^"]'
-if [ "${PIPESTATUS[1]}" -eq 0 ] ;then
-    error_out "rados ls" "$rados_out" "Found one or more RADOS objects existing in a namespace."
+# NOTE: Each entry (line of output) of `rados ls --all` should be in
+# one of four formats depending on whether or not an entry has a
+# namespace and/or locator:
+#
+#   <TAB>oid
+#   <TAB>oid<TAB>locator
+#   namespace<TAB>oid
+#   namespace<TAB>oid<TAB>locator
+#
+# Any occurrence of the 3rd or 4th (i.e., existence of namespace)
+# should cause this script to error out. Any occurence of the 2nd
+# (i.e., existence of locator) should generate a warning but allow the
+# script to continue to the end. We expect all entries to be of the
+# first line format.
+
+# check for namespaces and error out if any found; any line that
+# begins with graphical characters indicates a namespace as those
+# without a namespace will start with a tab
+grep '^[[:graph:]]' "$rados_out" >"$rados_odd"
+if [ "${PIPESTATUS[0]}" -eq 0 ] ;then
+    error_out "rados ls" "$rados_odd" "Found one or more RADOS objects existing in a namespace, which should not happen."
 fi
 
-# move from json to list of oids
-grep '^[[:blank:]]*"name":' "$rados_out" | sed 's/[^:]*: "\(.*\)"$/\1/' >"$temp_file"
+# look for locators; we idenitfy them by skipping past the empty
+# namespace, skipping past the oid, then looking for a TAB; note we
+# use egrep to get the '+' character and the $ in front of the '
+# allows the \t to be interpreted as a TAB
+egrep $'^[[:blank:]]+[[:graph:]]+\t' "$rados_out" >"$rados_odd"
+if [ "${PIPESTATUS[0]}" -eq 0 ] ;then
+    echo "*** WARNING: Some objects appear to use locators."
+    echo "***     see: $rados_odd"
+fi
+
+# extract just the oids for further processing; skip past the initial
+# blanks (for non-existant namespace), capture the graphics
+# characters, and skip past a possible tab and locator
+sed -E 's/^[[:blank:]]+([[:graph:]]+)(\t.*)?/\1/' "$rados_out" >"$temp_file"
```

Comment:
If we're stripping out the locator, but keeping the object name, will that make further processing error out on those entries from not being able to find the object?

Reply:
Theoretically there should not be locators. Yet they ended up in someone's cluster.

We need a list of just-OIDS from `rados ls ...` to compare against a list of just-OIDS from `radosgw-admin bucket radoslist ...`. The oid+d locator combinations are maintained in the *.issues output file for the benefit of the user. But you're right in that it creates extra work for the user in the end.